pkg/plugins/events/sns/sns.go

```diff
@@ -52,16 +54,26 @@ func (s *SnsEventService) getTopicArnFromName(name *string) (*string, error) {
 
 // Publish to a given topic
 func (s *SnsEventService) Publish(topic string, event *events.NitricEvent) error {
+	newErr := errors.ErrorsWithScope("SnsEventService.Publish")
+
 	data, err := json.Marshal(event)
 
 	if err != nil {
-		return fmt.Errorf("Payload marshalling error: %v", err)
+		return newErr(
+			codes.Internal,
+			"error marshalling event payload",
+			err,
+		)
 	}
 
 	topicArn, err := s.getTopicArnFromName(&topic)
 
 	if err != nil {
-		return fmt.Errorf("There was an error resolving the topic ARN for topic: %s, %v", topic, err)
+		return newErr(
+			codes.NotFound,
+			"could not find topic",
```

Comment:
was thinking about this, do you think it would be appropriate to include non-sensitive args in the error factory for scope?

update errors.ErrorsWithScope to be
```go
func ErrorsWithScope(scope string, args ...interface{})
```

e.g. newErr := errors.ErrorsWithScope("SnsEventService.Publish", topic)

and print them by default in error messages? so provided args will be included with error message by default?